examples/with-videojs/components/Player.js

```diff
@@ -1,34 +1,23 @@
-import { Component } from 'react'
+import { useRef, useEffect } from 'react'
 import videojs from 'video.js'
 import 'videojs-youtube'
 
-class Player extends Component {
-  componentDidMount() {
-    // instantiate Video.js
-    this.player = videojs(this.videoNode, this.props, function onPlayerReady() {
-      console.log('onPlayerReady', this)
-    })
-  }
+const Player = (props) => {
+  const playerRef = useRef()
```

Comment:
This needs to be `useCallback` AFAIK so that we are notified when the component mounts (to create the `videojs` instance).

Reply:
It works both in development and production right now, even if you close the page it will return you the video at the very same moment.

However, if you want me to modify it I would gladly do it.

Reply:
I suspect this works because of a race condition. Let's please make this use `useCallback` and `useState`.

Similar to this:
https://github.com/zeit/next.js/blob/canary/packages/react-dev-overlay/src/internal/components/LeftRightDialogHeader/LeftRightDialogHeader.tsx

Reply:
I have refactored the solution as you proposed, but I have fallen into the trap of another issue. Video.js uses Ref and since I am using React hooks I need to use useRef otherwise it wouldn't work.

The only thing that I could come up and that would work is something similar, but that doesn't actually improve anything related to your concerns.

![Screenshot 2020-05-26 at 20 24 02](https://user-images.githubusercontent.com/51530311/82930948-d838e080-9f8e-11ea-9193-0d6b920abebc.png)